TORCTL GUI

aleff-github opened the pull request:
In this pull request you can find a simple graphical interface to manage torctl, it is written in Python and the code is completely open source, it is released to the main repository under the GPLv3 license.

![GUI](https://raw.githubusercontent.com/aleff-github/torctl-gui/master/GUI/gui.png)

aleff-github committed:
TORCTL GUI

pwnfoo commented on GUI/gui.py:
```diff
@@ -0,0 +1,197 @@
+
+Licence: GPLv3"""
+
+CONTACT = """Github: https://github.com/aleff-github
```
Please don't include links to your personal socials in the code and claim ownership of torctl

aleff-github replied:
I had no intention of claiming ownership of torctl, I simply wanted to include a contact.

aleff-github replied:
Removed _Aleff thinks_, released only on About section on GUI Developer

pwnfoo commented on GUI/gui.py:
```diff
@@ -0,0 +1,197 @@
+# Create main window
+window = tk.Tk()
+window.title("TORCTL by Aleff")
+
```
Please don't take ownership of the project

aleff-github replied:
Replaced with `torctl - Redirect yout traffic`

pwnfoo commented on GUI/gui.py:
```diff
@@ -0,0 +1,197 @@
+def chngmac():
+    info_panel.configure(state="normal")
+    info_panel.insert(tk.END, "\n\nCHANGE MAC\n")
+    get_update_lines(subprocess.check_output("sudo torctl chngmac", shell=True))
```
Use absolute paths instead of "sudo command"

aleff-github replied:
Is this valid for any command?

aleff-github replied:
Replaced the `sudo` command with `/usr/local/bin/torctl`

pwnfoo commented on torctl:
```diff
@@ -15,7 +15,7 @@
 
 # torctl.sh version
-VERSION="torctl.sh v0.5.7"
+VERSION="torctl.sh v0.6.0"
```
wait for maintainers to bump version

aleff-github replied:
Ok so i don't change it?

aleff-github replied:
Undo update

aleff-github committed:
Suggestions by pwnfoo

- Personal contacts (I released only the GUI Developer contact (In the About section), hoping to keep it).
- Removed redundant code.
- Replaced the `sudo` command with `/usr/local/bin/torctl`

aleff-github commented:
@pwnfoo  I did what you recommended, however, I would like to point out that the absolute path to torctl does not preclude the need to use root permissions and thus the sudo command.

![example](https://i.ibb.co/VpRPbgP/Screenshot-from-2023-06-19-14-05-27.png)

aleff-github committed:
New Photo
